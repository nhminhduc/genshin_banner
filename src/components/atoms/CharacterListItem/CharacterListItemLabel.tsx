import cx from "classnames";
import { startCase } from "lodash";

interface CharacterListItemLabelProps {
  name: string;
  noVersions?: number;
}

const CharacterListItemLabel = ({
  name,
  noVersions,
}: CharacterListItemLabelProps) => (
  <div
    className={cx(
      "flex bg-[#e4c179] left-0 right-0 bottom-0 justify-between items-center h-8 px-1 py-2",
    )}
  >
    <div className={cx("font-bold text-sm text-slate-600")}>
      {startCase(name)}
    </div>
    {(noVersions || noVersions === 0) && (
      <div
        className={cx("font-bold text-xl", {
          "text-green-600": noVersions === 0,
          "text-lime-300": noVersions <= 3,
          "text-rose-600": noVersions > 3 && noVersions < 10,
          "text-rose-900": noVersions >= 10,
        })}
      >
        {noVersions}
      </div>
    )}
  </div>
);

export default CharacterListItemLabel;
