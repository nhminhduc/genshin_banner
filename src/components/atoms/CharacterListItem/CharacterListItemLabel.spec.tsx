import { render } from "@testing-library/react";
import { describe, expect, test } from "vitest";

import CharacterListItemLabel from "./CharacterListItemLabel";

describe("CharacterListItem test", () => {
  test("CharacterListItem snapshot", () => {
    const { container } = render(
      <CharacterListItemLabel name="CharacterListItem test" noVersions={4} />,
    );
    expect(container).toMatchSnapshot();
  });
});
