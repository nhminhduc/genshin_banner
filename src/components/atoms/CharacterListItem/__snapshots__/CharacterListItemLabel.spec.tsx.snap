// Vitest Snapshot v1

exports[`CharacterListItem test > CharacterListItem snapshot 1`] = `
<div>
  <div
    class="flex bg-[#e4c179] left-0 right-0 bottom-0 justify-between items-center h-8 px-1 py-2"
  >
    <div
      class="font-bold text-sm text-slate-600"
    >
      Character List Item Test
    </div>
    <div
      class="font-bold text-xl text-rose-600"
    >
      4
    </div>
  </div>
</div>
`;
